### PR TITLE
fix(ADRs): handle trailing slashes and collator errors

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

## Hey, I just made a Pull Request!

Small fixes for bugs in ADR plugin

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/adr-backend/src/search/DefaultAdrCollatorFactory.ts [hunk 1]
```diff
@@ -190,7 +190,10 @@ export class DefaultAdrCollatorFactory implements DocumentCollatorFactory {
       } catch (error: any) {
         // Ignore error if we're able to use cached response
         if (!cacheItem || error.name !== NotModifiedError.name) {
-          throw error;
+          this.logger.error(
+            `Unable to fetch ADRs from ${adrsUrl}: ${error.message}`,
```
Comment (freben) [inline plugins/adr-backend/src/search/DefaultAdrCollatorFactory.ts:194]: Could you use stringifyError instead of just the message?

**Review by freben — COMMENTED** [review]
Thanks! Just one minor nit

File: plugins/adr-backend/src/search/DefaultAdrCollatorFactory.ts (hunk 1 above)
Comment (kuangp) [reply plugins/adr-backend/src/search/DefaultAdrCollatorFactory.ts]: ooh neat, was not aware of this utility, updated!

**Review by freben — APPROVED** [review]
Nice!

**Pull request merged**